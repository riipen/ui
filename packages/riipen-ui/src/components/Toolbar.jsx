import clsx from "clsx";
import PropTypes from "prop-types";
import React from "react";

import withClasses from "../utils/withClasses";

class Toolbar extends React.Component {
  static propTypes = {
    /**
     * The content of the toolbar.
     */
    children: PropTypes.node,

    /**
     * List of additional classes to apply to this component.
     */
    classes: PropTypes.array
  };

  static defaultProps = {
    classes: []
  };

  render() {
    const { classes, children } = this.props;

    const className = clsx(classes);

    return (
      <React.Fragment>
        <div className={className}>{children}</div>
        <style jsx>{`
          div {
            align-items: center;
            display: flex;
            position: relative;
          }
        `}</style>
      </React.Fragment>
    );
  }
}

export default withClasses()(Toolbar);
