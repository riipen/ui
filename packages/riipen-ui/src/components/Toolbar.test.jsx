import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";

import Toolbar from "./Toolbar.jsx";

describe("<Toolbar>", () => {
  it("renders without errors", () => {
    let error;

    try {
      mount(<Toolbar />);
    } catch (e) {
      error = e;
    }

    expect(error).toEqual(undefined);
  });

  it("renders correct snapshot", () => {
    const wrapper = mount(<Toolbar />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe("children prop", () => {
    it("renders given children inside the div element", () => {
      const child = <h5>hello</h5>;

      const wrapper = mount(<Toolbar>{child}</Toolbar>);

      expect(
        wrapper
          .find("Toolbar")
          .find("div")
          .contains(child)
      ).toEqual(true);
    });
  });

  describe("classes prop", () => {
    it("applies all classes to the div inside of Toolbar", () => {
      const classes = ["classOne"];

      const wrapper = mount(<Toolbar classes={classes} />);

      expect(
        wrapper
          .find("Toolbar")
          .childAt(0)
          .find("div")
          .hasClass(classes[0])
      ).toEqual(true);
    });

    it("appends higher order values to classes prop with withClass decorator", () => {
      const classes = ["riipen", "riipen-toolbar"];

      const wrapper = mount(<Toolbar />);

      expect(
        wrapper
          .find("Toolbar")
          .props()
          .classes.sort()
      ).toEqual(classes.sort());
    });

    it("gives an error when classes are provided as one string", () => {
      const classes = "classOne classTwo";
      const errors = jest.spyOn(console, "error").mockImplementation();

      mount(<Toolbar classes={classes} />);

      expect(errors).toHaveBeenCalled();
    });
  });
});
