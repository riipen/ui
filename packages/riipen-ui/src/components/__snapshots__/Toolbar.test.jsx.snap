// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toolbar> renders correct snapshot 1`] = `
<withClasses(Toolbar)>
  <Toolbar
    classes={
      Array [
        "riipen",
        "riipen-toolbar",
      ]
    }
  >
    <div
      className="jsx-842450902 riipen riipen-toolbar"
    />
    <JSXStyle
      id="842450902"
    />
  </Toolbar>
</withClasses(Toolbar)>
`;
